File: airbyte-connector-builder-server/src/test/java/io/airbyte/connectorbuilder/commandrunner/MockAirbyteCdkProcess.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.commandrunner;

import java.io.IOException;

/**
 * Mock class for wrapping a process.
 */
public class MockAirbyteCdkProcess extends AirbyteCdkPythonProcess {

  private final Process process;

  public MockAirbyteCdkProcess(final Process process) {
    super(null, null, null, null);
    this.process = process;
  }

  @Override
  public Process start() throws IOException {
    return this.getProcess();
  }

  @Override
  public Process getProcess() {
    return this.process;
  }

  @Override
  public void close() {}

}

File: airbyte-connector-builder-server/src/test/kotlin/io/airbyte/connectorbuilder/commandrunner/MockAirbyteCdkProcess.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.commandrunner

import java.io.IOException

/**
 * Mock class for wrapping a process.
 */
class MockAirbyteCdkProcess(
  private val process: Process,
) : AirbyteCdkProcess {
  @Throws(IOException::class)
  override fun start(): Process = this.getProcess()

  override fun getProcess(): Process = this.process

  override fun close() {}
}
